src/client/opamClient.ml

```diff
@@ -635,12 +635,118 @@ let init_checks ?(hard_fail_exn=true) init_config =
   if hard_fail && hard_fail_exn then OpamStd.Sys.exit_because `Configuration_error
   else not (soft_fail || hard_fail)
 
-let windows_checks ?cygwin_setup config =
+let git_for_windows_check =
+  if not Sys.win32 && not Sys.cygwin then fun ?gitbin:_ () -> None else
+  fun ?gitbin () ->
+    let header () = OpamConsole.header_msg "Git" in
+    let contains_git p =
+      OpamSystem.resolve_command ~env:[||] (Filename.concat p "git.exe")
+    in
+    let gits =
+      OpamStd.Env.get "PATH"
+      |> OpamStd.Sys.split_path_variable
+      |> OpamStd.List.filter_map (fun p ->
+          match contains_git p with
+          | Some git ->
+            Some (git, OpamSystem.bin_contains_bash p)
+          | None -> None)
+    in
+    let get_gitbin ?gitbin () =
+      let bin =
+        match gitbin with
+        | Some _ -> gitbin
+        | None ->
+          OpamConsole.read "Enter winget or Git for Windows binary path (\\bin):"
+      in
+      match bin with
+      | None -> None
+      | Some gitbin ->
+        match contains_git gitbin, OpamSystem.bin_contains_bash gitbin with
+        | Some _, false ->
+          OpamConsole.msg "Using Git from %s" gitbin;
+          Some gitbin
+        | Some _, true ->
+          OpamConsole.error
+            "A bash executable was found in %s, which will override \
+             usual bash. Please check you binary path."
+            gitbin;
+          None
+        | None, _ ->
+          OpamConsole.error "No Git executable found in %s." gitbin;
+          None
+    in
+    let rec loop ?gitbin () =
+      match get_gitbin ?gitbin () with
+      | Some _ as gitbin -> gitbin
+      | None -> menu ()
+    and menu () =
+      let prompt () =
+        let options =
+          (`Default, "Use default Cygwin Git")
+          :: (List.filter_map (fun (git, bash) ->
+              if bash then None else
+              let bin = Filename.dirname git in
+              Some (`Location bin, "Use found git in "^bin))
+              gits)
+          @ [
+            `Specify, "Enter the location of installed Git";
+            `Abort, "Abort initialisation to install recommended Git.";
+          ]
+        in
+        OpamConsole.menu "Which Git should opam use?"
+          ~default:`Default ~no:`Default ~options
+      in
+      match prompt () with
+      | `Default -> None
+      | `Specify -> loop ()
+      | `Location gitbin -> loop ~gitbin ()
+      | `Abort ->
+        OpamConsole.note "Once your choosen Git installed, relaunch opam init.";
```

Comment:
```suggestion
        OpamConsole.note "Once your choosen Git installed, open a new PowerShell or Command Prompt window, and relaunch opam init.";
```

because even after a successful `winget install Git.Git` the prompt keeps coming back. I suspect that Git is not visible in the PATH of the same PowerShell window.

Reply:
Yes, indeed. There are a couple of solutions, if one is assuming that the user really installed Git:
- Refresh `PATH` from the Registry, but I don't recommend that we do that. Git for Windows setup will (or should) have broadcast [`WM_SETTINGCHANGE`](https://learn.microsoft.com/en-us/windows/win32/winmsg/wm-settingchange) which is the indicator to do this, the PATH environment can then be reloaded from HKEY_CURRENT_USER\Environment and HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment, but this is really fiddly and it doesn't respect that the user may have overridden PATH.
- Read the InstallPath value from HKEY_LOCAL_MACHINE\SOFTWARE\GitForWindows - I haven't double-checked what happens with a user-only installation, but after a system-wide installation, that contains the install location of Git for Windows